WCFDataService/Client/System/Data/Services/Client/MaterializeFromAtom.cs

```diff
@@ -103,6 +108,7 @@ internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
             this.expectingPrimitiveValue = PrimitiveType.IsKnownNullableType(elementType);
 
             Debug.Assert(responseMessage != null, "Response message is null! Did you mean to use Materializer.ResultsWrapper/EmptyResults?");
+            this.responseMessage = responseMessage;
```

Comment:
Then what about other places? Seems responseMessage will be passed to lots of places, for example, BatchResult...